### PR TITLE
Fix/1030

### PR DESCRIPTION
**Pull request by temehi**
This fixes readRecord() of SeqIO FormattedFile for FastQ files(issue #1030) by checking if there is nothing in between the next recored and the current quality string.

### CODE REVIEW

**Comment by h-2** [comment]
@temehi can you please investigate the test failures? Thanks!

**Comment by esiragusa** [comment]
@seqan/collaborators  Please always title your PR with a short description of the changes and eventually provide full description in the body of the PR. Otherwise you will have to prepare the next release changelog :trollface: 

@temehi Would you please describe which component is this PR fixing? Is it readRecord() of SeqIO FormattedFile?

File: include/seqan/seq_io/fasta_fastq.h [hunk 1]
```diff
@@ -413,7 +416,11 @@ inline void readRecord(TIdString & meta, TSeqString & seq, TQualString & qual, T
     TQualIgnoreOrAssert qualIgnore;
 
     readUntil(qual, iter, qualCountDown, qualIgnore);  // read Fastq qualities
-    skipUntil(iter, TFastqBegin());     // forward to the next '@'
+    skipUntil(iter, qualCountDown);                     // skip Fastq qualities
+    skipUntil(iter, NotFunctor<IsWhitespace>());        // skip Fastq qualities
```
Comment (h-2) [inline include/seqan/seq_io/fasta_fastq.h]: I don't understand the first `skipUntil()`. Why are you skipping if you are reading before? Please also fix the comments!

File: include/seqan/seq_io/fasta_fastq.h [hunk 2]
```diff
@@ -375,7 +375,10 @@ inline void readRecord(TIdString & meta, TSeqString & seq, TFwdIterator & iter,
     {
         skipUntil(iter, qualCountDown);                     // skip Fastq qualities
     }
-    skipUntil(iter, TFastqBegin());                         // forward to the next '@'
+    skipUntil(iter, NotFunctor<IsWhitespace>());        // skip Fastq qualities
+    TFastqBegin fastqBegin;
+    if(!fastqBegin(*(iter)) && *(iter) != '\xff' )  // check if next record follows
+        throw ParseError("quality line(s) have different length");
 }
```
Comment (h-2) [inline include/seqan/seq_io/fasta_fastq.h]: Description is not very clear, maybe instead:
"Fastq quality string expected to be of same length as sequence string, but was not."

Comment (temehi) [inline include/seqan/seq_io/fasta_fastq.h]: Noted!

**Comment by h-2** [comment]
@temehi  good catch! 

But please don't do whitespace changes in the lines that you are not changing. I think you changed the space between command and comment. This might sound like :nail_care: but it makes the diff harder to read, because it looks like more changes...

**Comment by temehi** [comment]
> I think you changed the space between command and comment. 

Yes! You are right, but in my defense I did it to get the comments aligned without making the lines too long. This might make the diff hard to read but on tge other hand it  makes the code itself more  readable.

**Comment by h-2** [comment]
@temehi still please try to keep the diffs minimal. Especially line 376 should not be changed.

**Comment by h-2** [comment]
@temehi can you please fix the last request, i.e. no whitespace changes? Thanks!

**Pull request merged**